### PR TITLE
Feature/match detail

### PR DESCRIPTION
**Pull request by RuddyKingfisher**

### CODE REVIEW

**Comment by eddy-kor-92** [comment]
compile error 나는거 아닌가요?

**Comment by eddy-kor-92** [comment]
|File|Coverage [21.96%]|:x:|
|:-|:-:|:-:|
|[TypeDefinitions.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fentities%2FTypeDefinitions.kt)|93.79%|:green_apple:|
|[SummonerService.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FSummonerService.kt)|77.21%|:x:|
|[SummonerDTO.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Fdto%2Fsummoners%2FSummonerDTO.kt)|66.67%|:x:|
|[Match.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fentities%2FMatch.kt)|61.98%|:x:|
|[MatchService.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FMatchService.kt)|15.25%|:x:|
|[PageDto.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Fdto%2Fcommon%2FPageDto.kt)|0%|:x:|
|[MatchDto.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FMatchDto.kt)|0%|:x:|
|[Champions.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FChampions.kt)|0%|:x:|
|[TeamColor.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FTeamColor.kt)|0%|:x:|
|[DataDragon.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FDataDragon.kt)|0%|:x:|
|[SummonerSpells.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FSummonerSpells.kt)|0%|:x:|
|[MatchType.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FMatchType.kt)|0%|:x:|
|[SummonerController.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Frest%2FSummonerController.kt)|0%|:x:|
|[MatchController.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Frest%2FMatchController.kt)|0%|:x:|
|[TestController.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Frest%2FTestController.kt)|0%|:x:|
|[WebConstants.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Fconstants%2FWebConstants.kt)|0%|:x:|
|[JpaSummonerRepository.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fjpa%2FJpaSummonerRepository.kt)|0%|:x:|
|[JdbcSummonerRepository.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fjdbc%2FJdbcSummonerRepository.kt)|0%|:x:|
|[DataDragonService.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FDataDragonService.kt)|0%|:x:|
|[ChampionService.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FChampionService.kt)|0%|:x:|
|[SummonerSpellsService.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FSummonerSpellsService.kt)|0%|:x:|
|[RiotApiService.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FRiotApiService.kt)|0%|:x:|
|[MatchSummonerRepositoryImpl.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fimpl%2FMatchSummonerRepositoryImpl.kt)|0%|:x:|
|[MatchRepositoryImpl.kt](https://github.com/SOTATER/back-end/blob/cd50153b003c94edb50cc3a5bf9dbc264ef7b67a/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fimpl%2FMatchRepositoryImpl.kt)|0%|:x:|

|Total Project Coverage|25.23%|:x:|
|:-|:-:|:-:|

**Comment by eddy-kor-92** [comment]
|File|Coverage [22.01%]|:x:|
|:-|:-:|:-:|
|[TypeDefinitions.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fentities%2FTypeDefinitions.kt)|93.79%|:green_apple:|
|[SummonerService.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FSummonerService.kt)|77.21%|:x:|
|[SummonerDTO.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Fdto%2Fsummoners%2FSummonerDTO.kt)|66.67%|:x:|
|[Match.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fentities%2FMatch.kt)|61.98%|:x:|
|[MatchService.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FMatchService.kt)|15.25%|:x:|
|[PageDto.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Fdto%2Fcommon%2FPageDto.kt)|0%|:x:|
|[MatchDto.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FMatchDto.kt)|0%|:x:|
|[Champions.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FChampions.kt)|0%|:x:|
|[TeamColor.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FTeamColor.kt)|0%|:x:|
|[DataDragon.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FDataDragon.kt)|0%|:x:|
|[SummonerSpells.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FSummonerSpells.kt)|0%|:x:|
|[MatchType.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fdto%2FMatchType.kt)|0%|:x:|
|[SummonerController.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Frest%2FSummonerController.kt)|0%|:x:|
|[MatchController.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Frest%2FMatchController.kt)|0%|:x:|
|[TestController.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Frest%2FTestController.kt)|0%|:x:|
|[WebConstants.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fweb%2Fconstants%2FWebConstants.kt)|0%|:x:|
|[JpaSummonerRepository.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fjpa%2FJpaSummonerRepository.kt)|0%|:x:|
|[JdbcSummonerRepository.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fjdbc%2FJdbcSummonerRepository.kt)|0%|:x:|
|[DataDragonService.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FDataDragonService.kt)|0%|:x:|
|[ChampionService.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FChampionService.kt)|0%|:x:|
|[SummonerSpellsService.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FSummonerSpellsService.kt)|0%|:x:|
|[RiotApiService.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdomain%2Fservices%2FRiotApiService.kt)|0%|:x:|
|[MatchSummonerRepositoryImpl.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fimpl%2FMatchSummonerRepositoryImpl.kt)|0%|:x:|
|[MatchRepositoryImpl.kt](https://github.com/SOTATER/back-end/blob/807712564bbe8b1ebdc5a74597e937dab83a70ce/src%2Fmain%2Fkotlin%2Fcom%2Fsota%2Fclone%2Fcopyopgg%2Fdatabase%2Fimpl%2FMatchRepositoryImpl.kt)|0%|:x:|

|Total Project Coverage|25.27%|:x:|
|:-|:-:|:-:|

File: src/main/kotlin/com/sota/clone/copyopgg/web/rest/TestController.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.sota.clone.copyopgg.domain.dto.DataDragonSummonerSpellData
+import com.sota.clone.copyopgg.domain.services.DataDragonService
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/api/test")
+class TestController(
```
Comment (eddy-kor-92) [inline src/main/kotlin/com/sota/clone/copyopgg/web/rest/TestController.kt:13]: Test controller는 뭔가요?

File: src/main/kotlin/com/sota/clone/copyopgg/domain/entities/TypeDefinitions.kt [hunk 2]
```diff
@@ -16,7 +16,7 @@ enum class GameType {
 enum class GameMode {
     ARAM,
     ARSR,
-    ASCENSION,
+    ASCENSION(),
```
[the remark this answers is not in the document]
Comment (RuddyKingfisher) [reply src/main/kotlin/com/sota/clone/copyopgg/domain/entities/TypeDefinitions.kt:19]: 실수로 들어간 것 같네요

File: src/main/kotlin/com/sota/clone/copyopgg/web/rest/TestController.kt (hunk 1 above)
Comment (RuddyKingfisher) [reply src/main/kotlin/com/sota/clone/copyopgg/web/rest/TestController.kt:13]: 요거 그냥 테스트용으로 사용할 API를 위해서 만든 클래스입니다

**Pull request merged**